integration/int_3Ri_2BhaRi2_3Re_2BhaRe3/test_multicast_basic.py

```diff
@@ -0,0 +1,202 @@
+    #in this case no consumer is accepting ...
+    #{
+     #"recv_outcomes": [Outcome.modify, Outcome.release],
+     #"expected": Expected.accepted,
+     #"test_id": "Expect accept if no rejects=====",
+    #},
+    #{
+     #"recv_outcomes": 3*[Outcome.release] + [Outcome.modify],
+     #"expected": Expected.modified,
+     #"test_id": "Expect modified over released=====",
+    #},
+    #{
+     #"recv_outcomes": 2*[Outcome.modify],
+     #"expected": Expected.modified,
+     #"test_id": "Expected modify if all modify",
+    #},
+    {
+     "recv_outcomes": 6*[Outcome.release],
+     "expected": Expected.released,
+     "test_id": "Release only if all released",
+    },
+]
+
+@pytest.fixture(params=outcomes_config_list, ids=idfn)
+def outcomes(request):
+    return request.param
+
+class _Receiver(Receiver):
+    def __init__(self, *args, settle=Outcome.accept, **kwargs):
+        super(_Receiver, self).__init__(*args, auto_accept=False, **kwargs)
+        self._settle = getattr(self, settle)
+
+    def modify(self, delivery):
+        super(_Receiver, self).release(delivery, delivered=True)
+
+    def release(self, delivery):
+        super(_Receiver, self).release(delivery, delivered=False)
+
+    def on_message(self, event):
+        self.last_received_id[event.message.user_id] = event.message.id
+        self.received += 1
+        self.messages.append(event.message.body)
+
+        logging.info("settle = %s" % self._settle.__name__)
+        self._settle(event.delivery)
+
+        if self.is_done_receiving():
+            self.stop_receiver(event.receiver, event.connection)
+
+class _Sender(Sender):
+    def is_done_sending(self):
+        return (self.stopped or (self.total > 0 and self.sent == self.total))
+
+class TestMulticast:
+    MESSAGES_COUNT = 5
+    MESSAGE_SIZE = 128
+
+    TIMEOUT = 4 #why?
+    address = "multicast/bla"
+
+    @staticmethod
+    def _get_router_url(router, topic):
+        return "amqp://%s:%s/%s" % (router.node.get_ip(), router.port, topic)
+
+    def _sender(self, router, topic):
+        s = _Sender(url=self._get_router_url(router, topic),
+                   message_count=self.MESSAGES_COUNT,
+                   sender_id='sender-%s' % router.node.hostname,
+                   timeout=self.TIMEOUT,
+                   message_size=self.MESSAGE_SIZE,
+                   use_unique_body=True,
+                   auto_settle=False,
+                  )
+
+        s.start()
+        return s
+
+    def _receiver(self, router, topic, settle=Outcome.modify):
+        r = _Receiver(url=self._get_router_url(router, topic),
+                      message_count=self.MESSAGES_COUNT,
+                      settle=settle,
+                      timeout=self.TIMEOUT,
+                      )
+        r.start()
+        return r
+
+    def launch_receivers(self, outcomes, iqa):
+        def _wait(receivers):
+            for r in receivers:
+                while not r.receiver:
+                    time.sleep(1)
```

Comment:
Not sure if a time.sleep here might cause issues to proton.
Maybe we can use some of the proton events to consider the receivers in the list as ready.